### PR TITLE
feat(gatsby): add telemetry for usage of lazy devjs bundling

### PR DESCRIPTION
**Pull request by KyleAMathews**
Didn't make it into the original PR https://github.com/gatsbyjs/gatsby/pull/27884/

### CODE REVIEW

File: packages/gatsby/src/services/initialize.ts [hunk 1]
```diff
@@ -177,6 +177,9 @@ export async function initialize({
       telemetry.trackFeatureIsUsed(`QueryOnDemand`)
     }
   }
+  if (process.env.GATSBY_EXPERIMENT_DEVJS_LAZY) {
+    telemetry.trackFeatureIsUsed(`lazy-dev-js-bundling`)
```
Comment (vladar) [inline packages/gatsby/src/services/initialize.ts:181]: Let's agree on naming conventions? With this PR we will have 3 different styles as we also have this usage:
https://github.com/gatsbyjs/gatsby/blob/a612f26cab94a7672f6c63067368ec6efc49ec49/packages/gatsby/src/utils/dev-ssr/develop-html-route.ts#L10

I like the `GATSBY_EXPERIMENTAL_DEV_SSR` way of naming as it also matches the flag name which is great for "greppability"?

Can we use `GATSBY_EXPERIMENT_DEVJS_LAZY` and `GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND` in this file for feature names? CC @pieh 

Also looks like the flag name for lazy devjs is also inconsistent 😄 It should be `GATSBY_EXPERIMENTAL_DEVJS_LAZY`

Comment (KyleAMathews) [reply packages/gatsby/src/services/initialize.ts]: haha yeah I got confused. We got to get an `experiments.js` file for this stuff.

Comment (jamo) [reply packages/gatsby/src/services/initialize.ts]: All current tracked features are in PascalCase so maybe we could use that convention also here.

Comment (jamo) [reply packages/gatsby/src/services/initialize.ts]: So it would be `GatsbyExperimentalDevSsr` or just `ExperimentalDevSsr` or `ExperimentalDevSSR`

Comment (vladar) [reply packages/gatsby/src/services/initialize.ts]: I didn't realize we already use this feature in some plugins, e.g.:

https://github.com/gatsbyjs/gatsby/blob/9e6091325485bbcbcfa04bccda09170e383ae395/packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts#L40

 So what @jamo suggests makes sense for consistency. Also probably need to update the name for Dev SSR https://github.com/gatsbyjs/gatsby/blob/a612f26cab94a7672f6c63067368ec6efc49ec49/packages/gatsby/src/utils/dev-ssr/develop-html-route.ts#L10

Comment (KyleAMathews) [reply packages/gatsby/src/services/initialize.ts]: 👍 making change

**Review by jamo — APPROVED** [review]

**Pull request merged**